### PR TITLE
Dynamic splitter

### PR DESCRIPTION
**Pull request by gmaggiodev**
### Feature Requests
With this change I implemented dynamic management of panels into splitter.
In this way you can support more dynamic situations like creating a dashboard that applow creating dynamically the panels.

For example (this can be also found in the splitter documentation where I added an example in the bottom of the page) you can bind a panel structure (array in this case) and manipulating the model the UI react as expected.

```       
<p-splitter [style]="{ height: '300px' }" styleClass="mb-5">
                <ng-container *ngFor="let item of items">
                    <ng-template pTemplate>
                        <div class="col flex align-items-center justify-content-center">{{item}}</div>
                    </ng-template>
                </ng-container>
 </p-splitter>
```

https://github.com/primefaces/primeng/assets/1485446/cf7ecb83-6afc-4397-91c7-8b785ddb4479

### CODE REVIEW

File: src/app/components/splitter/splitter.ts [hunk 1]
```diff
@@ -158,16 +158,13 @@ export class Splitter {
...
-            switch (item.getType()) {
-                case 'panel':
-                    this.panels.push(item.template);
-                    break;
-                default:
-                    this.panels.push(item.template);
-                    break;
-            }
+
+        this.templates.changes.subscribe(_ => {
```
Comment (volvachev) [inline src/app/components/splitter/splitter.ts:162]: Hello, please add an unsubscribe.
Why do you use `detectChanges` and not `markForCheck`?

File: src/app/components/splitter/splitter.ts [hunk 2]
```diff
@@ -158,16 +158,13 @@ export class Splitter {
...
+            this.initPanels();
+            this.cd.detectChanges();
         });
+
+        this.initPanels();
```
Comment (volvachev) [inline src/app/components/splitter/splitter.ts:167]: what about this variant and remove `this.initPanels();` outside pipe.

```ts
this.templates.changes.pipe(
  startWith(null),
  ...unsubscribe
).subscribe(() => {
 this.initPanels();
 this.cd.markForCheck();
})
```

File: src/app/components/splitter/splitter.ts [hunk 3]
```diff
@@ -193,6 +190,20 @@ export class Splitter {
         }
     }
 
+    initPanels() {
+        this.panels = [];
+        this.templates.forEach((item) => {
```
Comment (volvachev) [inline src/app/components/splitter/splitter.ts:195]: Is it better this way?
```ts
this.panels = this.templates.reduce((acc, item) => {
     switch (item.getType()) {
         case 'panel':
             acc.push(item.template);
              break;
         default:
             acc.push(item.template);
              break;
    }
    return acc;
}, []);
```

File: src/app/components/splitter/splitter.ts (hunk 2 above)
Comment (gmaggiodev) [reply src/app/components/splitter/splitter.ts:167]: yes, it's better but unsubscribe should go into ngOnDestroy to preserve the dynamicity. I'll check it for a new commit.

**Comment by volvachev** [comment]
> @volvachev do you think is better to remove the package-lock.json? I notice a lot of differences.

Yep, remove pls

**Comment by gmaggiodev** [comment]
Testing better the dynamic case I found a glitch. After adding the new panel, using the gutter to resize the panel has a litte "jump". This video can show you the defect.


https://github.com/primefaces/primeng/assets/1485446/e20568b9-8ae7-4b14-970d-a8bfb69d3070

Not sure but probably depends on the fact that panelSizes are not updated after adding the new panel.

@volvachev can you help me?

**Comment by gmaggiodev** [comment]
Hi @volvachev just added sizes calculation in case of dynamic panels. Let me know.
Thanks

**Comment by gmaggiodev** [comment]
@volvachev any news?

**Comment by cetincakiroglu** [comment]
Hi @gmaggiodev ,

Thank you for taking the time to submit your pull request. We genuinely appreciate your effort and interest in improving our project.

After careful review and consideration, we have decided not to merge your pull request at this time. Please understand that this decision does not reflect negatively on the quality of your work. We value every contribution and understand the effort you've put into it.

However, upon review, we believe that the changes proposed in this pull request may not align with the current goals or direction of the project. Our team is continuously evaluating and prioritizing features and enhancements based on various factors, including project roadmap, user feedback, and technical considerations.

We encourage you to continue contributing to the project and to stay engaged with our community. Your input and contributions are valuable, and we look forward to seeing more of your work in the future.

Once again, thank you for your contribution, and please don't hesitate to reach out if you have any questions or would like further feedback.

Best regards,

**Pull request closed without merging**